enhancements/console/customize-catalogs.md

```diff
@@ -0,0 +1,210 @@
+the catalog or the sub-catalog to streamline their 
+development process without uninstalling a complete operator or disabling the complete console plugin.
+
+In the developer catalog the sub-catalogs are shown as types. 
+The developer catalog shows items from all sub-catalogs.
+When user selects a sub-catalog, they can see a list of 
+available items from this sub-catalog to install.
+
+This enhancement allows cluster admins to selectively 
+disable a sub-catalog and all links to the specific sub-catalog. 
+The developer catalog will also hide any of the items from this sub-catalog.
+
+![Developer catalog and types list](customize-catalogs.png)
+ 
+## Motivation
+
+The developer catalog can be extended by plugins and 
+also dynamically by installing a console plugin or an operator.
+
+Some customers want to align their development process and 
+have teams that support other development teams with their work.
+For this cluster administrators want to hide developer sub-catalog (types) 
+without disabling a full console plugin, for example by uninstalling an operator.
+By hiding for example "Service Backed operators", "Helm", etc. 
+they can still install them via the CLI, see their resources in the topology, 
+but don't show them in the add from developer catalog flow.
+
+### User Stories
+
+#### Story 1
+
+Customer wants to display only their own customized and validated 
+templates on the dev console but they can't change the default devfiles currently.
+
+As a cluster admin, I want to hide Developer catalog for all users.
+For example for customers who has their own way to 
+install workloads, services, etc. without the developer catalog.
+
+#### Story 2
+
+Customer wants to display only their own customized and validated 
+templates on the dev console but they can't change the default devfiles currently.
+
+As a cluster admin, I want to hide certain sub-catalogs (types) for all users.
+For example for customers who don't want to use Helm charts, 
+or another sub-catalog without disabling Helm support in the whole developer console.
+
+### Goals
+
+- Allow cluster admins to hide the developer catalog or certain sub-catalogs (types).
+- Provide information to admins about sub-catalogs (types) they can hide from users.
+
+### Non-Goals
+
+- Allow cluster admins to add enabled sub-catalogs (types) in customization. 
+- Allow cluster admins to customize categories or hide individual items.
+
+## Proposal
+
+We can extend the existing `operator.openshift.io/v1` / `Console CRD`. 
+We can add 
+`spec.customization.developerCatalog.disabledTypes` which lists types to hide.
+
+### Workflow Description
+
+```text
+├── spec
+│   ├── customization
+|       ├── developerCatalog 
+|           ├── disabledTypes
+|               └──       
+└── ...
+```
+
+Each catalog conforms to the following schema:
+
+// +optional
+- `disabledTypes: string[]` , defines a list of types to hide. 
+Types will vary based on the operator installed.
+
+### API Extensions
```

Comment:
Please add the proposed Go structure for the operator changes so that we can review that here as well

Reply:
The PR is already up here https://github.com/openshift/api/pull/1256, if really needed we can add this here as well.